### PR TITLE
Remove the concept of epochs and dropping of peers from the lightning module

### PR DESCRIPTION
**Pull request by joschisan**

### CODE REVIEW

File: modules/fedimint-ln-server/src/lib.rs [hunk 1]
```diff
@@ -855,16 +840,29 @@ impl Lightning {
...
+
+        let mut heights = vec![];
+
+        for index in 0..peer_count {
+            heights.push(
+                dbtx.get_value(&BlockHeightVoteKey(PeerId::from(index as u16)))
+                    .await
+                    .unwrap_or(0),
+            )
         }
```
Comment (elsirion) [inline modules/fedimint-ln-server/src/lib.rs:861]: Could this be a single call to `dbtx.find_by_prefix` instead?

File: modules/fedimint-ln-server/src/lib.rs [hunk 2]
```diff
@@ -855,16 +840,29 @@ impl Lightning {
         Ok(Lightning { cfg, btc_rpc })
     }
 
-    /// Returns the median block height if enough peers have contributed
-    pub async fn consensus_height(&self, dbtx: &mut ModuleDatabaseTransaction<'_>) -> Option<u64> {
-        let peer_heights = dbtx.get_value(&BlockHeightKey).await.unwrap_or_default();
-        if peer_heights.len() < self.cfg.consensus.threshold() {
-            return None;
+    pub async fn block_height(&self) -> u64 {
```
Comment (elsirion) [inline modules/fedimint-ln-server/src/lib.rs:843]: This function seems a bit dangerous, accidental use in consensus logic would break consensus.

**Review by elsirion — CHANGES_REQUESTED** [review]
Some minor nits, one DoS vector I'd like to discuss/solve before merging.

File: modules/fedimint-ln-common/src/db.rs [hunk 3]
```diff
@@ -125,11 +127,18 @@ impl_db_lookup!(
 );
 
 #[derive(Debug, Encodable, Decodable, Serialize)]
-pub struct BlockHeightKey;
+pub struct BlockHeightVoteKey(pub PeerId);
```
Comment (m1sterc001guy) [inline modules/fedimint-ln-common/src/db.rs:130]: Do we want this to be backwards compatible? This will require a database migration

Comment (elsirion) [reply modules/fedimint-ln-common/src/db.rs:130]: Probably not until we do a release.

File: modules/fedimint-ln-server/src/lib.rs (hunk 1 above)
Comment (joschisan) [reply modules/fedimint-ln-server/src/lib.rs]: Done

File: modules/fedimint-ln-server/src/lib.rs (hunk 2 above)
Comment (joschisan) [reply modules/fedimint-ln-server/src/lib.rs:841]: We could rename it to local_block_height or something?

**Review by elsirion — APPROVED** [review]
I'm ok with this version, maybe you can squeeze in the rename in the next PR as a separate commit :)

**Pull request merged**